etl/steps/data/garden/energy/2022-12-12/owid_energy.py

```diff
@@ -0,0 +1,211 @@
+        variable_metadata = variable_mapping.loc[source_variable]
+        source_dataset = variable_metadata["source_dataset"]
+        # Check that the variable indeed exists in the original dataset that the variable mapping says.
+        # Ignore columns "country", "year" (assigned to a dummy dataset 'various_datasets'), "population" (that comes
+        # from key_indicators) and "iso_alpha3" (that comes from countries_regions dataset).
+        if source_dataset not in [
+            "various_datasets",
+            "countries_regions",
+            "key_indicators",
+            "maddison_gdp",
+        ]:
+            error = f"Variable {source_variable} not found in any of the original datasets."
+            assert source_variable in tables[source_dataset].columns, error
+            tb_combined[source_variable].metadata = tables[source_dataset][source_variable].metadata
+
+        # Update metadata with the content of the variable mapping file.
+        tb_combined[source_variable].metadata.description = variable_metadata["description"]
+        tb_combined[source_variable].metadata.sources = [catalog.meta.Source(name=variable_metadata["source"])]
+
+    # Select only variables in the mapping file, and rename variables according to the mapping.
+    tb_combined = tb_combined[source_variables].rename(columns=variable_mapping.to_dict()["variable"])
+
+    # Remove rows that only have nan (ignoring if country, year, iso_code, population and gdp do have data).
+    columns_that_must_have_data = [
+        column for column in tb_combined.columns if column not in ["country", "year", "iso_code", "population", "gdp"]
+    ]
+    tb_combined = tb_combined.dropna(subset=columns_that_must_have_data, how="all").reset_index(drop=True)
+
+    # Sanity check.
+    columns_with_inf = [column for column in tb_combined.columns if len(tb_combined[tb_combined[column] == np.inf]) > 0]
+    assert len(columns_with_inf) == 0, f"Infinity values detected in columns: {columns_with_inf}"
+
+    # Set index and sort conveniently.
+    tb_combined = tb_combined.set_index(["country", "year"], verify_integrity=True).sort_index()
+
+    return cast(catalog.Table, tb_combined)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load data.
+    #
+    # Read all required datasets.
+    ds_energy_mix = catalog.Dataset(ENERGY_MIX_DATASET_PATH)
+    ds_fossil_fuels = catalog.Dataset(FOSSIL_FUEL_PRODUCTION_DATASET_PATH)
+    ds_primary_energy = catalog.Dataset(PRIMARY_ENERGY_CONSUMPTION_DATASET_PATH)
+    ds_electricity_mix = catalog.Dataset(ELECTRICITY_MIX_DATASET_PATH)
+
+    # Gather all required tables from all datasets.
+    tb_energy_mix = ds_energy_mix[ds_energy_mix.table_names[0]].reset_index()
+    tb_fossil_fuels = ds_fossil_fuels[ds_fossil_fuels.table_names[0]].reset_index()
+    tb_primary_energy = ds_primary_energy[ds_primary_energy.table_names[0]].reset_index()
+    tb_electricity_mix = ds_electricity_mix[ds_electricity_mix.table_names[0]].reset_index()
```

Comment:
I would rather define the specific names of the tables you expect to load (instead of assuming the first one is the one).